server/src/test/java/com/witness/server/integration/web/BaseControllerIntegrationTest.java

```diff
@@ -108,7 +108,7 @@ protected String requestUrl() {
   }
 
   protected String requestUrl(String suffix, Object... args) {
-    return requestUrl() + "/" + String.format(suffix, args);
+    return StringUtils.trimTrailingCharacter(requestUrl() + "/" + String.format(suffix, args), '/');
```

Comment:
**very** important to know: https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide#spring-mvc-and-webflux-url-matching-changes

I have played around with the client in an emulator and tested the most important features by hand - it seems that we only send "correct" URLs from the Android app to the server. However, just so that you know, should somewhere/some time an unexpected "404 Not Found" be returned, this might be the cause. But I am relatively confident that we have implemented the client correctly.